### PR TITLE
Create nouvelles-perspectives-en-sciences-sociales.csl

### PR DESCRIPTION
**Pull request by lsaadaoui**
npss

### CODE REVIEW

File: nouvelles-perspectives-des-sciences-sociales.csl [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+    <choose>
+      <if type="song" match="all" variable="composer">
+        <names variable="composer">
+          <name and="text" delimiter-precedes-last="never" initialize="false" name-as-sort-order="all"/>
+        </names>
+      </if>
+      <else>
+        <names variable="author">
+          <name and="text" delimiter-precedes-last="never" initialize="false" name-as-sort-order="all">
+            <name-part name="family"/>
```
Comment (adam3smith) [inline nouvelles-perspectives-des-sciences-sociales.csl:49]: what's your goal in specifying the name-part here? The way you have it here, without any styling (like `text-case` of `font-style`) it doesn't do anything.  (Same for all further instances below)

File: nouvelles-perspectives-des-sciences-sociales.csl [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+  <macro name="translator">
+    <names variable="translator">
+      <label form="verb" suffix=" "/>
+      <name and="text" delimiter-precedes-last="never" initialize="false" sort-separator=" "/>
+    </names>
+  </macro>
+  <macro name="editor">
+    <group>
+      <names variable="editor" delimiter=", ">
+        <name and="text" delimiter-precedes-last="never" initialize="false" name-as-sort-order="all">
```
Comment (adam3smith) [inline nouvelles-perspectives-des-sciences-sociales.csl:115]: no need for initialize="false" without `initialize-with`

File: nouvelles-perspectives-des-sciences-sociales.csl [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+            </else>
+          </choose>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="publisher">
+    <choose>
+      <if type="manuscript graphic legislation legal_case bill article-journal article-magazine article-newspaper review-book interview speech treaty dataset" match="none">
+        <group delimiter=", ">
```
Comment (adam3smith) [inline nouvelles-perspectives-des-sciences-sociales.csl:188]: this group doesn't do anything

**Review by adam3smith — CHANGES_REQUESTED** [review]
Looks good! A couple of small clean-up requests, including one question

**Comment by rmzelle** [comment]
Thanks! Did you base this on an existing CSL style? If so, we'd like to know which one, so we can add a "template" link (see https://github.com/citation-style-language/styles/wiki/Style-Requirements#8---template-link)

**Review by lsaadaoui — COMMENTED** [review]
Please delete csl style nouvelles-perspectives-des-sciences-sociales.csl #5063

**Review by lsaadaoui — COMMENTED** [review]
The good csl style for nouvelles-perspectives-des-sciences-sociales.csl was created by Ines-dot. Please can you delete mine (#5063) and keep the good one #5109 ?

File: nouvelles-perspectives-des-sciences-sociales.csl (hunk 1 above)
Comment (Ines-dot) [reply nouvelles-perspectives-des-sciences-sociales.csl]: We are Following the bibliography of the journal http://npssrevue.ca/en/

File: nouvelles-perspectives-des-sciences-sociales.csl (hunk 3 above)
Comment (Ines-dot) [reply nouvelles-perspectives-des-sciences-sociales.csl]: We are sorry. We will make all the suggested corrections

File: nouvelles-perspectives-en-sciences-sociales.csl [hunk 4]
```diff
@@ -2,17 +2,19 @@
 <style xmlns="http://purl.org/net/xbiblio/csl" class="note" version="1.0" and="text" initialize="false" demote-non-dropping-particle="never" default-locale="fr-CA">
   <!-- This style was edited with the Visual CSL Editor (https://editor.citationstyles.org/visualEditor/) -->
   <info>
-    <title>Nouvelles Perspectives des Sciences Sociales (French - Canada)</title>
+    <title>Nouvelles perspectives en sciences sociales (French - Canada)</title>
```
Comment (Ines-dot) [inline nouvelles-perspectives-en-sciences-sociales.csl]: approved

File: nouvelles-perspectives-en-sciences-sociales.csl [hunk 5]
```diff
@@ -2,17 +2,19 @@
...
     <title-short>NPSS</title-short>
-    <id>http://www.zotero.org/styles/nouvelles-perspectives-des-sciences-sociales</id>
-    <link href="http://www.zotero.org/styles/nouvelles-perspectives-des-sciences-sociales" rel="self"/>
+    <id>http://www.zotero.org/styles/nouvelles-perspectives-en-sciences-sociales</id>
```
Comment (Ines-dot) [inline nouvelles-perspectives-en-sciences-sociales.csl]: approved

File: nouvelles-perspectives-en-sciences-sociales.csl [hunk 6]
```diff
@@ -24,10 +26,10 @@
       <term name="translator" form="verb-short">trad. par</term>
       <term name="translator" form="short">traduction</term>
       <term name="interviewer" form="verb">entretien réalisé par</term>
-      <term name="in">dans </term>
+      <term name="in">dans&#160;</term>
```
Comment (Ines-dot) [inline nouvelles-perspectives-en-sciences-sociales.csl]: &#160;

File: nouvelles-perspectives-en-sciences-sociales.csl [hunk 7]
```diff
@@ -2,17 +2,19 @@
 <style xmlns="http://purl.org/net/xbiblio/csl" class="note" version="1.0" and="text" initialize="false" demote-non-dropping-particle="never" default-locale="fr-CA">
   <!-- This style was edited with the Visual CSL Editor (https://editor.citationstyles.org/visualEditor/) -->
   <info>
-    <title>Nouvelles Perspectives des Sciences Sociales (French - Canada)</title>
+    <title>Nouvelles perspectives en sciences sociales (French - Canada)</title>
     <title-short>NPSS</title-short>
-    <id>http://www.zotero.org/styles/nouvelles-perspectives-des-sciences-sociales</id>
-    <link href="http://www.zotero.org/styles/nouvelles-perspectives-des-sciences-sociales" rel="self"/>
+    <id>http://www.zotero.org/styles/nouvelles-perspectives-en-sciences-sociales</id>
+    <link href="http://www.zotero.org/styles/nouvelles-perspectives-en-sciences-sociales" rel="self"/>
```
Comment (Ines-dot) [inline nouvelles-perspectives-en-sciences-sociales.csl]: approved

File: nouvelles-perspectives-en-sciences-sociales.csl [hunk 8]
```diff
@@ -24,10 +26,10 @@
       <term name="translator" form="verb-short">trad. par</term>
       <term name="translator" form="short">traduction</term>
       <term name="interviewer" form="verb">entretien réalisé par</term>
-      <term name="in">dans </term>
+      <term name="in">dans&#160;</term>
       <term name="edition">édition</term>
       <term name="accessed">consulté le</term>
-      <term name="at">disponible sur :</term>
+      <term name="at">disponible sur&#160;:</term>
```
Comment (Ines-dot) [inline nouvelles-perspectives-en-sciences-sociales.csl]: approved

File: nouvelles-perspectives-en-sciences-sociales.csl [hunk 9]
```diff
@@ -524,7 +526,7 @@
                 <text value="loc. cit." font-style="italic"/>
               </if>
               <else-if type="book report thesis chapter entry-encyclopedia entry-dictionary" match="any">
-                <text value="op. cit." font-style="italic"/>
+                <text value="op.&#160;cit." font-style="italic"/>
```
Comment (Ines-dot) [inline nouvelles-perspectives-en-sciences-sociales.csl]: approved

File: nouvelles-perspectives-en-sciences-sociales.csl [hunk 10]
```diff
@@ -536,7 +538,7 @@
       </choose>
     </layout>
   </citation>
-  <bibliography delimiter-precedes-last="never" et-al-min="11" et-al-use-first="7" initialize="false" subsequent-author-substitute="———" hanging-indent="true">
+  <bibliography delimiter-precedes-last="never" et-al-min="11" et-al-use-first="7" initialize="false" subsequent-author-substitute="&#8212;&#8212;&#8212;" hanging-indent="true">
```
Comment (Ines-dot) [inline nouvelles-perspectives-en-sciences-sociales.csl]: approved

**Review by Ines-dot — APPROVED** [review]

**Review by Ines-dot — APPROVED** [review]

**Review by Ines-dot — APPROVED** [review]

**Review by Ines-dot — APPROVED** [review]

File: nouvelles-perspectives-en-sciences-sociales.csl (hunk 4 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-en-sciences-sociales.csl]: Approved

File: nouvelles-perspectives-en-sciences-sociales.csl (hunk 5 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-en-sciences-sociales.csl]: We approve this change

File: nouvelles-perspectives-en-sciences-sociales.csl (hunk 7 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-en-sciences-sociales.csl]: We approve this change

File: nouvelles-perspectives-en-sciences-sociales.csl (hunk 4 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-en-sciences-sociales.csl]: We approve this change. The correct title is : Nouvelles perspectives en sciences sociales

File: nouvelles-perspectives-des-sciences-sociales.csl (hunk 3 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-des-sciences-sociales.csl]: The comma will not appear ? Do we have to cancel it ?

File: nouvelles-perspectives-des-sciences-sociales.csl (hunk 1 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-des-sciences-sociales.csl]: For wich line ?

File: nouvelles-perspectives-des-sciences-sociales.csl (hunk 2 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-des-sciences-sociales.csl]: OK

**Review by lsaadaoui — COMMENTED** [review]
We approve the modification in the title

File: nouvelles-perspectives-en-sciences-sociales.csl (hunk 6 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-en-sciences-sociales.csl]: This modification will not affect our proposition ? We need a bibliography for our journal as the sample citations that csl-bot sent us. The citations generated and based on our proposed changes.

File: nouvelles-perspectives-en-sciences-sociales.csl (hunk 8 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-en-sciences-sociales.csl]: This modification will not affect our proposition ? We need a bibliography for our journal as the sample citations that csl-bot sent us. The citations generated and based on our proposed changes.

File: nouvelles-perspectives-en-sciences-sociales.csl (hunk 9 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-en-sciences-sociales.csl]: This modification will not affect our proposition ? We need a bibliography for our journal as the sample citations that csl-bot sent us. The citations generated and based on our proposed changes.

File: nouvelles-perspectives-en-sciences-sociales.csl (hunk 10 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-en-sciences-sociales.csl]: This modification will not affect our proposition ? We need a bibliography for our journal as the sample citations that csl-bot sent us. The citations generated and based on our proposed changes.

File: nouvelles-perspectives-des-sciences-sociales.csl [hunk 11]
```diff
@@ -6,7 +6,6 @@
     <title-short>NPSS</title-short>
     <id>http://www.zotero.org/styles/nouvelles-perspectives-des-sciences-sociales</id>
     <link href="http://www.zotero.org/styles/nouvelles-perspectives-des-sciences-sociales" rel="self"/>
-    <link href="http://www.zotero.org/styles/npssrevue" rel="template"/>
```
Comment (lsaadaoui) [inline nouvelles-perspectives-des-sciences-sociales.csl]: ok

**Review by lsaadaoui — COMMENTED** [review]
We want to add (French - Canada) to the title.

**Review by lsaadaoui — COMMENTED** [review]
[Nouvelles Perspectives en Sciences Sociales (French - Canada ).txt](https://github.com/citation-style-language/styles/files/5593424/Nouvelles.Perspectives.en.Sciences.Sociales.French.-.Canada.txt)

**Review by Ines-dot — APPROVED** [review]

File: nouvelles-perspectives-des-sciences-sociales.csl [hunk 12]
```diff
@@ -0,0 +1,606 @@
+<?xml version="1.0" encoding="utf-8"?>
+<style xmlns="http://purl.org/net/xbiblio/csl" class="note" version="1.0" and="text" initialize="false" demote-non-dropping-particle="never" default-locale="fr-CA">
+  <!-- This style was edited with the Visual CSL Editor (https://editor.citationstyles.org/visualEditor/) -->
+  <info>
+    <title>Nouvelles Perspectives des Sciences Sociales (French - Canada)</title>
+    <title-short>NPSS</title-short>
+    <id>http://www.zotero.org/styles/nouvelles-perspectives-des-sciences-sociales</id>
+    <link href="http://www.zotero.org/styles/nouvelles-perspectives-des-sciences-sociales" rel="self"/>
```
Comment (Ines-dot) [inline nouvelles-perspectives-des-sciences-sociales.csl]: <link href="http://www.zotero.org/styles/nouvelles-perspectives-en-sciences-sociales" rel="self"/>

File: nouvelles-perspectives-des-sciences-sociales.csl [hunk 13]
```diff
@@ -0,0 +1,606 @@
+<?xml version="1.0" encoding="utf-8"?>
+<style xmlns="http://purl.org/net/xbiblio/csl" class="note" version="1.0" and="text" initialize="false" demote-non-dropping-particle="never" default-locale="fr-CA">
+  <!-- This style was edited with the Visual CSL Editor (https://editor.citationstyles.org/visualEditor/) -->
+  <info>
+    <title>Nouvelles Perspectives des Sciences Sociales (French - Canada)</title>
```
Comment (Ines-dot) [inline nouvelles-perspectives-des-sciences-sociales.csl]: <title>Nouvelles Perspectives en Sciences Sociales (French - Canada)</title>

File: nouvelles-perspectives-des-sciences-sociales.csl [hunk 14]
```diff
@@ -0,0 +1,606 @@
...
+    <title-short>NPSS</title-short>
+    <id>http://www.zotero.org/styles/nouvelles-perspectives-des-sciences-sociales</id>
```
Comment (Ines-dot) [inline nouvelles-perspectives-des-sciences-sociales.csl]: <id>http://www.zotero.org/styles/nouvelles-perspectives-en-sciences-sociales</id>

**Review by Ines-dot — APPROVED** [review]

**Comment by lsaadaoui** [comment]
> Thanks! Did you base this on an existing CSL style? If so, we'd like to know which one, so we can add a "template" link (see https://github.com/citation-style-language/styles/wiki/Style-Requirements#8---template-link)

This is the "instructions to authors" webpage of Nouvelles perspectives en sciences sociales: http://npssrevue.ca/guide/biblio/

**Comment by rmzelle** [comment]
> > Thanks! Did you base this on an existing CSL style? If so, we'd like to know which one, so we can add a "template" link (see https://github.com/citation-style-language/styles/wiki/Style-Requirements#8---template-link)
> 
> This is the "instructions to authors" webpage of Nouvelles perspectives en sciences sociales: http://npssrevue.ca/guide/biblio/

No, sorry, you're misunderstanding me.

It's rare for CSL styles (the XML files like the one you're submitting here) to be written from scratch. Most CSL style authors take an existing CSL style and modify it as needed to match their preferred format.

Was the NPSS CSL style written from scratch, or, like most CSL styles, derived from another CSL style? If it was derived, we'd like to keep track what CSL style you started from.

File: nouvelles-perspectives-en-sciences-sociales.csl (hunk 6 above)
Comment (rmzelle) [reply nouvelles-perspectives-en-sciences-sociales.csl:29]: @lsaadaoui, no, `&#160;` is just a different (escaped) XML representation of a non-breaking space. We just prefer to see the special character codes as visually these non-breaking spaces are indistinguishable from regular spaces. It doesn't change how the CSL style behaves.

File: nouvelles-perspectives-en-sciences-sociales.csl (hunk 10 above)
Comment (rmzelle) [reply nouvelles-perspectives-en-sciences-sociales.csl:541]: @lsaadaoui, same here. The various hyphen and dash characters are hard to tell apart, so we use character codes for en-dashes and em-dashes.

**Comment by lsaadaoui** [comment]
> > > Thanks! Did you base this on an existing CSL style? If so, we'd like to know which one, so we can add a "template" link (see https://github.com/citation-style-language/styles/wiki/Style-Requirements#8---template-link)
> > 
> > 
> > This is the "instructions to authors" webpage of Nouvelles perspectives en sciences sociales: http://npssrevue.ca/guide/biblio/
> 
> No, sorry, you're misunderstanding me.
> 
> It's rare for CSL styles (the XML files like the one you're submitting here) to be written from scratch. Most CSL style authors take an existing CSL style and modify it as needed to match their preferred format.
> 
> Was the NPSS CSL style written from scratch, or, like most CSL styles, derived from another CSL style? If it was derived, we'd like to keep track what CSL style you started from.

Thanks for the clarifications. The NPSS CSL style was derived from the style of Université du Québec à Montréal - Études littéraires et sémiologie (French - Canada)

File: nouvelles-perspectives-en-sciences-sociales.csl (hunk 6 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-en-sciences-sociales.csl:29]: Thanks for the clarifications

File: nouvelles-perspectives-en-sciences-sociales.csl (hunk 10 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-en-sciences-sociales.csl:541]: Thanks for the clarifications

File: nouvelles-perspectives-des-sciences-sociales.csl (hunk 2 above)
Comment (lsaadaoui) [reply nouvelles-perspectives-des-sciences-sociales.csl]: This modification will not affect our proposition ? We need a bibliography for our journal as the sample citations that csl-bot sent us. The citations generated and based on our proposed changes.

**Comment by rmzelle** [comment]
> Thanks for the clarifications. The NPSS CSL style was derived from the style of Université du Québec à Montréal - Études littéraires et sémiologie (French - Canada)

Great, thanks. I incorporated @adam3smith suggestions and merged the style. Thanks for your contribution!

**Comment by lsaadaoui** [comment]
> > Thanks for the clarifications. The NPSS CSL style was derived from the style of Université du Québec à Montréal - Études littéraires et sémiologie (French - Canada)
> 
> Great, thanks. I incorporated @adam3smith's suggestions and merged the style. Thanks for your contribution!

Thanks a lot for your help. The NPSS CSL style will be in Zotero Style Repository ?

File: nouvelles-perspectives-des-sciences-sociales.csl (hunk 2 above)
Comment (rmzelle) [reply nouvelles-perspectives-des-sciences-sociales.csl]: It shouldn't, no.

That said, if you discover you need to make any changes to the style, please follow the instructions at https://github.com/citation-style-language/styles/blob/master/CONTRIBUTING.md#3b-submitting-changes-to-an-existing-style.

**Comment by rmzelle** [comment]
> The NPSS CSL style will be in Zotero Style Repository ?

We're currently making some infrastructure changes, so it might be a few days until the Zotero Style Repository updates.

**Comment by lsaadaoui** [comment]
> We're currently making some infrastructure changes to this GitHub repository, so it might be a few days until the Zotero Style Repository updates.

Thanks a lot for all the clarifications. We will keep an eye on the Zotero Style Repository.

**Comment by rmzelle** [comment]
This style can now be found in the Zotero Style Repository: https://www.zotero.org/styles?q=id%3Anouvelles-perspectives-en-sciences-sociales

**Pull request merged**